### ISSUE TITLE
Building on Windows

### ISSUE THREAD

**Issue by rupipal** [cisco/ChezScheme#155]
I downloaded zip from https://cisco.github.io/ChezScheme/#get   Extracted it. Following https://github.com/cisco/ChezScheme/blob/master/BUILDING for windows, I installed Wix toolset  and extension for Visual Studio 2015 that I have. cygwin is also installed and in Path. I renamed configure to configure.exe and ran it.  Nothing seems to happen. command shell doesn't do anything.

**Comment by okuoku** [comment]
> cygwin is also installed and in Path. I renamed configure to configure.exe and ran it. 

You don't have to do so. Steps should be:

1. Open "VS2015 x64 Native Tools Command Prompt"
2. On the "VS2015 x64 Native Tools Command Prompt" opened, start Cygwin with `c:\cygwin64\Cygwin.bat`
3. (At this point, it would be nice to check cl.exe on the path; `which cl` -- "Native Tools" should configure `PATH` value for you so you don't have to change `PATH` by yourself)
4. checkout Git repository / extract .zip somewhere and `cd` into it
5. run `./configure` 
6. run `make`

In a nutshell, you will have to build ChezScheme on Cygwin shell; it does not provide Visual Studio project file. Don't forget installing `make`, `git` and other friends on your Cygwin install.

**Comment by okuoku** [comment]
(Step2 should look like this.)

![snapcrab_noname_2017-4-2_18-9-37_no-00](https://cloud.githubusercontent.com/assets/100813/24585866/8f1f2ac0-17cf-11e7-9d4d-cacc9f8dcd27.png)

**Comment by rupipal** [comment]
Hi,
First of all, I have 32-bit system. So I installed Native tools prompt accordingly.

which cl and which make didn't show after Native tools and so I added the bin carrying those to PATH varaible. Even after adding, which cl and which make did't show, but I went ahead. The output of the remaining steps is as follows:
nwe@nwe-PC /cygdrive/d/swbuilds/ChezScheme-9.4
$ ./configure
no suitable machine type found
try rerunning as ./configure -m=<machine type>
available machine types: a6le, a6nt, a6osx, i3le, i3nt, i3osx, ta6le, ta6nt, ta6
osx, ti3le, ti3nt, and ti3osx

nwe@nwe-PC /cygdrive/d/swbuilds/ChezScheme-9.4
$ ./configure -m=i3nt
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   140    0   140    0     0     65      0 --:--:--  0:00:02 --:--:--    84
  0     0    0 3726k    0     0  36146      0 --:--:--  0:01:45 --:--:-- 58062
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   119    0   119    0     0     59      0 --:--:--  0:00:02 --:--:--    77
100  590k  100  590k    0     0  28594      0  0:00:21  0:00:21 --:--:-- 23863
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   119    0   119    0     0     64      0 --:--:--  0:00:01 --:--:--    79
  0     0    0  206k    0     0  18299      0 --:--:--  0:00:11 --:--:-- 27083

nwe@nwe-PC /cygdrive/d/swbuilds/ChezScheme-9.4
$
I tried to run scheme.exe  and petite.exe from within git repository. They say Access is denied. I guess it isn't built properly.

Rupinder

**Comment by okuoku** [comment]
.zip distribution failed to me too.. Perhaps it would be better to use git source since it worked for me, both `i3nt` and `ti3nt`. You don't have to install Chez prior building git sources.

> which cl and which make didn't show after Native tools and so I added the bin carrying those to PATH varaible.

You don't have to fiddle with `PATH` variable. The script should find correct tools based on environment variables. VS2015 installer should set `VS120COMNTOOLS` variable and [the script](https://github.com/cisco/ChezScheme/blob/master/c/Mf-i3nt) uses it.

... actually I haven't aware they've fixed the script in the last year( https://github.com/cisco/ChezScheme/commit/11ac1ba5a8fc3f16b997c498d223328f73b31152 ).. Now my steps above are just redundant.

1. Open Cygwin (Adding Cygwin tools to `PATH` do not work here. Always use Cygwin.bat or `mintty` short cut which Cygwin installed to invoke `bash` prompt and issue every command there.)
2. `./configure -m=i3nt`
3. `make`

that's it.

> I tried to run scheme.exe and petite.exe from within git repository. They say Access is denied. I guess it isn't built properly.

Perhaps it is just a symlink to somewhere. It won't work outside of Cygwin.

I put my build at http://storage.osdev.info/pub/proj/yuni/Chez%20Scheme.msi I believe it is the expected output (built with VS2015pro).

**Comment by rupipal** [comment]
Hi.

The installer works, of course.

However, I get the following output on building from source. I cleaned the installation directory once and re-cloned. It persists.

nwe@nwe-PC /cygdrive/d/swbuilds/ChezScheme
$ ./configure -m=i3nt
./configure: line 2: $'\r': command not found
./configure: line 17: $'\r': command not found
./configure: line 18: $'\r': command not found
./configure: line 19: syntax error near unexpected token `$'do\r''
'/configure: line 19: `for fn in boot/*/scheme.boot ; do

I get the same output with -m=ti3nt
Rupinder

**Comment by burgerrg** [comment]
This looks like a CRLF line ending problem. Are you using Cygwin's git? Is it configured to use CRLF line endings? If so, configure it not to change the line endings.

**Comment by rupipal** [comment]
Thanks okuoku and Bob!

Despite okuoku's reminder, I forgot to install git and make on Cygwin, which I did now and build seems ok. Again I forgot to give argument -m=i3nt, but make seems to have detected it. To be safe, re-ran with -m=i3nt. Both scheme.exe and petite.exe are available and run ok. I am attaching the complete ouput for record.
[ouput.txt](https://github.com/cisco/ChezScheme/files/894834/ouput.txt)

 
Rupinder

**Comment by rupipal** [comment]
I am closing this issue by summarizing Okuoku's and Bob's suggestions for use by the other prospective beginners .
1. Ensure that Cygwin is installed on your Windows with git and make packages. 

2. Open Cygwin by Cygwin shortcut (minty) or Cygwin.bat

3. cd to some directory you want to install in.

4. Clone current ChezScheme source from https://github.com/cisco/ChezScheme.git to that directory using 
git clone https://github.com/cisco/ChezScheme.git 
-- a git repository will be created named ChezScheme.

5. cd to ChezScheme directory cloned as above.

6.  ./configure -m=i3nt 

7.  make

8. scheme.exe and petite.exe should be available in newly created folders i3nt. 

9. Repeat steps 6. and 7. with -m=ti3nt, if you want threaded version too.
